modules/bullet/bullet_physics_server.cpp

```diff
@@ -1462,24 +1526,29 @@ void BulletPhysicsServer3D::init() {
 }
 
 void BulletPhysicsServer3D::step(real_t p_deltaTime) {
+#ifndef _3D_DISABLED
+
 	if (!active) {
 		return;
 	}
 
 	BulletPhysicsDirectBodyState3D::singleton_setDeltaTime(p_deltaTime);
 
-	for (int i = 0; i < active_spaces_count; ++i) {
-		active_spaces[i]->step(p_deltaTime);
+	for (Set<SpaceBullet *>::Element *E = active_spaces.front(); E; E = E->next()) {
+		SpaceBullet *space = (SpaceBullet *)E->get();
+		space->step(p_deltaTime);
 	}
+
+#endif
 }
 
 void BulletPhysicsServer3D::flush_queries() {
 	if (!active) {
 		return;
 	}
-
-	for (int i = 0; i < active_spaces_count; ++i) {
-		active_spaces[i]->flush_queries();
+	for (Set<SpaceBullet *>::Element *E = active_spaces.front(); E; E = E->next()) {
+		SpaceBullet *space = (SpaceBullet *)E->get();
```

Comment:
I'm surprised that an explicit cast would be needed (here and in other places using `Set<SpaceBullet *>::Element`). Doesn't the template guarantee that `E->get()` *is* a `SpaceBullet *` (or `nullptr` I guess)?

Reply:
I just was taken that from the default physics (to make it uniform)
![image](https://user-images.githubusercontent.com/76991284/115796129-91877c80-a3d1-11eb-9534-fc687f158153.png)

Reply:
![image](https://user-images.githubusercontent.com/76991284/115796192-afed7800-a3d1-11eb-8c4f-db79564ec8e7.png)

I just was taken that from the default physics (to make it uniform)

Reply:
As commented above, this doesn't seem to need a set, so it can be reverted.

Reply:
Small note, better to avoid C cast, since the compiler is unable to catch some potential error. Much better use `static_cast`, `dynamic_cast`, etc...